### PR TITLE
Use references/1 when generating migrations in ecto model task

### PR DESCRIPTION
**Pull request by gjaldon**
Closes https://github.com/phoenixframework/phoenix/issues/1006

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -40,6 +40,7 @@ defmodule Phoenix.Mixfile do
 
      # Test dependencies
      {:phoenix_html, "~> 1.2", only: :test},
+     {:ecto, "~> 0.14.1", only: :test},
```
Comment (josevalim) [inline mix.exs]: Yeah, we should not depend on Ecto. Please just export the __schema__(:source) function from a blank module.

File: test/mix/tasks/phoenix.gen.model_test.exs [hunk 2]
```diff
@@ -3,6 +3,14 @@ Code.require_file "../../../installer/test/mix_helper.exs", __DIR__
 defmodule Phoenix.Dup do
 end
 
+defmodule Phoenix.Web do
```
Comment (josevalim) [inline test/mix/tasks/phoenix.gen.model_test.exs]: Let's nuke this to keep the test simple, just define the model inline here.

File: test/mix/tasks/phoenix.gen.model_test.exs [hunk 3]
```diff
@@ -103,6 +112,18 @@ defmodule Mix.Tasks.Phoenix.Gen.ModelTest do
         assert file =~ "field :title, :string"
         assert file =~ "belongs_to :user, Phoenix.User"
       end
+
+      assert_raise Mix.Error, fn ->
```
Comment (josevalim) [inline test/mix/tasks/phoenix.gen.model_test.exs]: Please define a separate test where you simply use a dummy module defined in this file. :)

File: lib/mix/tasks/phoenix.gen.model.ex [hunk 4]
```diff
@@ -94,15 +94,40 @@ defmodule Mix.Tasks.Phoenix.Gen.Model do
...
+  end
+
+  defp get_assoc_source(key) do
+    assoc  = Mix.Phoenix.inflect Atom.to_string(key)
+    module = Module.concat(Elixir, assoc[:module])
+    if Code.ensure_loaded?(module) do
+      module.__schema__(:source) |> String.to_atom()
+    else
+      Mix.raise """
+      The table name for the association is inferred from the assocation
```
Comment (josevalim) [inline lib/mix/tasks/phoenix.gen.model.ex]: This error message is great although we don't say which module we could not load. Something like:

    The table name for the association is inferred from the assocation
    module but could not load #{inspect module}. You will need to explicitly
    set the association's table similar to:

**Comment by josevalim** [comment]
This is exactly what I had in mind to solve that issue! thank @gjaldon, I have added some comments!

**Comment by gjaldon** [comment]
@josevalim great to hear that and appreciate the inputs. Will apply those changes now. :)

**Comment by ericmj** [comment]
Can we avoid the ecto dependency?

**Comment by gjaldon** [comment]
@ericmj yes, we can. I'll be removing it in the next commit.

File: mix.lock [hunk 5]
```diff
@@ -1,6 +1,8 @@
 %{"cowboy": {:hex, :cowboy, "1.0.0"},
   "cowlib": {:hex, :cowlib, "1.0.1"},
+  "decimal": {:hex, :decimal, "1.1.0"},
   "earmark": {:hex, :earmark, "0.1.17"},
+  "ecto": {:hex, :ecto, "0.14.1"},
```
Comment (josevalim) [inline mix.lock]: `mix deps.unlock ecto` :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: mix.lock (hunk 5 above)
Comment (gjaldon) [inline mix.lock]: Ooops :D

**Pull request merged**